common/deliver/deliver.go

```diff
@@ -260,6 +271,13 @@ func (h *Handler) deliverBlocks(ctx context.Context, srv *Server, envelope *cb.E
 
 	logger.Debugf("[channel: %s] Received seekInfo (%p) %v from %s", chdr.ChannelId, seekInfo, seekInfo, addr)
 
+	// Throttling limits the number of os threads created when getting blocks (file i/o's) from ledger.
+	if err := h.ThrottleSemaphore.Acquire(ctx); err != nil {
+		logger.Errorf("[channel: %s] Failed to acquire semaphore to get a block", chdr.ChannelId)
+		return cb.Status_BAD_REQUEST, errors.WithMessage(err, "failed to acquire semaphore to get a block")
```

Comment:
I'm not sure that this should be Status_BAD_REQUEST , shouldn't this be Status_INTERNAL_SERVER_ERROR ?

Reply:
If semaphore.Acquire returns an error, it means the user context is cancelled (e.g., client closed the connection). Therefore I return a bad request, but looking at the existing code, it actually returns Status_INTERNAL_SERVER_ERROR. Will change to Status_INTERNAL_SERVER_ERROR.